app/portainer/home/EnvironmentList/KubeconfigButton.tsx

```diff
@@ -20,9 +34,42 @@ export function KubeconfigButton({ environments }: Props) {
   }
 
   return (
-    <Button onClick={handleClick}>
-      <i className="fas fa-download space-right" /> kubeconfig
-    </Button>
+    <div>
+      <Button onClick={handleClick}>
+        <i className="fas fa-download space-right" /> kubeconfig
+      </Button>
+      <DialogOverlay
+        className="center-dialog"
+        isOpen={showDialog}
+        aria-label="Kubeconfig View"
+        role="dialog"
+      >
+        <div className="modal-content">
+          <div className="modal-header">
+            <button
+              type="button"
+              className="bootbox-close-button close"
+              aria-hidden="true"
+              onClick={handleClose}
+            >
+              ×
+            </button>
+            <h5 className="modal-title">Download kubeconfig file</h5>
+          </div>
+          <KubeconfigPormpt
+            ksExpiry={ksExpiry}
```

Comment:
please use full name (not sure what `ks` stands for)

Reply:
Changed to kubeService